docs/formatting.md

```diff
@@ -170,45 +225,101 @@ one (254814599) two (656421459) three (-1117028319)
 
-new Rectangle
-{   
-    Height = 50, 
-    Width = 100
+A formatter source can also specify the preferred MIME types for your type:
+
+```csharp
+[TypeFormatterSource(
+  typeof(MyFormatterSource), 
+  PreferredMimeTypes = new[] { "text/html", "application/json" })]
+public class MyTypeWithCustomFormatting
+{
 }
 ```
 
-```console
-Rectangle
-  Location: Point
-    IsEmpty: True
-    X: 0
-    Y: 0
-  Size: Size
-    IsEmpty: False
-    Width: 100
-    Height: 50
-  X: 0
-  Y: 0
-  Width: 100
-  Height: 50
-  Left: 0
-  Top: 0
-  Right: 100
-  Bottom: 50
-  IsEmpty: False
+Specifying several MIME types results in a call to `Formatter.SetPreferredMimeTypesFor`, as described under [Preferred MIME types](#preferred-mime-types). The behavior is equivalent.
+
+If you don't want to take a dependency on `Microsoft.DotNet.Interactive.Formatting`, there's another to way to provide custom formatting for your types. This can be the simpler approach if you're building a library and you'd like to provide an augmented experience in notebooks but you don't need the other tools available to .NET Interactive extensions, such as magic commands and custom kernels.
+
+With this approach, you declare your own `TypeFormatterSourceAttribute` and apply it. The shape of the class should match the `TypeFormatterSourceAttribute` defined in `Microsoft.DotNet.Interactive.Formatting`. The `TypeFormatterSourceAttribute` class that you define will be recognized by name. Like the formatter source, it can be declared as internal so that it has no impact on your public API surface.
+
+Here's the complete definition, which you can copy into your project:
+
+```csharp
+[AttributeUsage(AttributeTargets.Class)]
+internal class TypeFormatterSourceAttribute : Attribute
+{
+    public TypeFormatterSourceAttribute(Type formatterSourceType)
+    {
+        FormatterSourceType = formatterSourceType;
+    }
+
+    public Type FormatterSourceType { get; }
+
+    public string[] PreferredMimeTypes { get; set; }
+}
+```
+
+As with the attribute class itself, the specified formatter source and the formatters it returns are based on naming conventions rather than types. The following is a complete example that you can use as a template:
+
+```csharp
+internal class MyConventionBasedFormatterSource
+{
+    public IEnumerable<object> CreateTypeFormatters()
+    {
+        yield return new MyConventionBasedFormatter { MimeType = "text/html" };
+    }
+}
+
+internal class MyConventionBasedFormatter
+{
+    public string MimeType { get; set; }
+
+    public bool Format(object instance, TextWriter writer)
+    {
+        if (instance is MyTypeWithCustomFormatting myObj)
+        {
+            writer.Write($"<div>Custom formattering for {obj}</div>");
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+}
```

Comment:
It may be a good idea to repeat the following snippet for the convention-based registration sample as well (for completeness / to avoid confusion).

```csharp
[TypeFormatterSource(typeof(MyFormatterSource))]
public class MyTypeWithCustomFormatting
{
}
```